// window.ai/dist/index.cjs
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.ts
var src_exports = {};
__export(src_exports, {
  ErrorCode: () => ErrorCode,
  EventType: () => EventType,
  VALID_DOMAIN: () => VALID_DOMAIN,
  getWindowAI: () => getWindowAI,
  hasWindowAI: () => hasWindowAI,
  isMessageOutput: () => isMessageOutput,
  isMessagesInput: () => isMessagesInput,
  isTextOutput: () => isTextOutput,
  waitForWindowAI: () => waitForWindowAI
});
module.exports = __toCommonJS(src_exports);
function isMessagesInput(input) {
  return "messages" in input;
}
function isTextOutput(output) {
  return "text" in output;
}
function isMessageOutput(output) {
  return "message" in output;
}
var ErrorCode = /* @__PURE__ */ ((ErrorCode2) => {
  ErrorCode2["NotAuthenticated"] = "NOT_AUTHENTICATED";
  ErrorCode2["PermissionDenied"] = "PERMISSION_DENIED";
  ErrorCode2["RequestNotFound"] = "REQUEST_NOT_FOUND";
  ErrorCode2["InvalidRequest"] = "INVALID_REQUEST";
  ErrorCode2["ModelRejectedRequest"] = "MODEL_REJECTED_REQUEST";
  return ErrorCode2;
})(ErrorCode || {});
var EventType = /* @__PURE__ */ ((EventType2) => {
  EventType2["ModelChanged"] = "model_changed";
  EventType2["Error"] = "error";
  return EventType2;
})(EventType || {});
var VALID_DOMAIN = "https://windowai.io";
function hasWindowAI() {
  return typeof globalThis.window.ai?.getCompletion === "function";
}
var DEFAULT_WAIT_OPTIONS = {
  interval: 100,
  timeout: 2400
  // https://github.com/alexanderatallah/window.ai/pull/34#discussion_r1170545022
};
async function waitForWindowAI(opts = DEFAULT_WAIT_OPTIONS) {
  if (hasWindowAI()) {
    return;
  }
  await new Promise((resolve, reject) => {
    let counter = 0;
    const timerInterval = setInterval(() => {
      counter += opts.interval;
      if (counter > opts.timeout) {
        clearInterval(timerInterval);
        reject(new Error("window.ai not found"));
      }
      if (hasWindowAI()) {
        clearInterval(timerInterval);
        resolve(true);
      }
    }, opts.interval);
  });
}
var getWindowAI = async (opts = DEFAULT_WAIT_OPTIONS) => {
  await waitForWindowAI(opts);
  return globalThis.window.ai;
};
